### PR TITLE
Create forgot password functionality

### PR DESCRIPTION
**Pull request by en3on**

### CODE REVIEW

File: controllers/auth-controller.js [hunk 1]
```diff
@@ -96,29 +100,19 @@ async function login(req, res, next) {
...
-    console.log({resetPassLink});
+    const resetPassLink = generateResetPassLink(user);
 
-    await User.findOneAndUpdate(
-        {email: user.email},
-        {resetPassLink: resetPassLink},
-        (err, doc) => {
-          if (err) console.log({err});
-        });
+    console.log('Saving resetpasslink');
```
Comment (PraveenMut) [inline controllers/auth-controller.js]: Not a big thing, but do we need these console logs in production? Could be good for debugging though, but yeah, just letting you know.

File: controllers/auth-controller.js [hunk 2]
```diff
@@ -96,29 +100,19 @@ async function login(req, res, next) {
...
 
-    const updatedUser = await User.findOne({email: user.email});
+    await resetPassLink.save();
 
-    console.log({updatedUser});
+    console.log('resetpasslink saved', {resetPassLink});
```
Comment (PraveenMut) [inline controllers/auth-controller.js]: see first comment

File: utils/error-utils.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const DEBUG = false;
+const DEBUG = true;
```
Comment (PraveenMut) [inline utils/error-utils.js]: if this is merged to master, will this need to be set as true?

**Review by PraveenMut — CHANGES_REQUESTED** [review]

File: controllers/auth-controller.js (hunk 1 above)
Comment (en3on) [reply controllers/auth-controller.js]: Hahaha thanks for the heads up! Forgot about those! Monday morning lol

Comment (en3on) [reply controllers/auth-controller.js]: fixed

File: controllers/auth-controller.js (hunk 2 above)
Comment (en3on) [reply controllers/auth-controller.js]: fixed

File: utils/error-utils.js (hunk 3 above)
Comment (en3on) [reply utils/error-utils.js]: fixed

**Review by PraveenMut — APPROVED** [review]
super nice! no qualms here! :)

**Pull request merged**